edge/key.go

```diff
@@ -21,26 +21,25 @@ type edgeKey struct {
 
 // SetKey parses and associates an Edge key to the agent.
 // If the agent is running inside a cluster, it will also set the "set" flag to specify that a key is set on this agent in the cluster.
-func (manager *Manager) SetKey(key string) error {
+func (manager *Manager) SetKey(key string, persist bool) error {
```

Comment:
That new persist parameter should be documented in a comment. I'm also not sure to understand why we have a specific case where the key is not persisted.

I can see:

* Key set via HTTP UI -> persisted
* Key set via API -> persisted
* Key set via environment -> not persisted

Why is that?

Reply:
The only time the key is not persisted is when it was already read from disk, to avoid the error when trying to write it again. Perhaps I should overwrite it atomically to avoid problems instead? What do you think?

Reply:
I'd say to try to remove the `persist` parameter, and just rewrite like Andres says, if some part of the flow needs to check wether to persist the key, it can be donde outside if this function

Reply:
>The only time the key is not persisted is when it was already read from disk

That's not entirely true, when the agent does a lookup for the key it will look sequentially:

* In the environment/cli flags
* On the filesystem
* In the cluster (if it's running a multi-agent deployment on Swarm/Kubernetes)

That's basically what the `RetrieveEdgeKey` function does, here you're assuming that each time this function is called it will find a key on the FS.

I would remove the persist flag and always persist at the moment. I'm actually thinking that persisting it on disk might be pointless and that instead we should only have a lookup:

* In the environment
* In the cluster 

Running the agent standalone would only land in the first case.
Cluster would always do 1 or 2 anyway. Plus that would get rid of statefulness.

Reply:
I am not assuming it will find a key, the code will call `SetKey(edgeKey, false)` only after succesfully retrieving the key from the filesystem: https://github.com/portainer/agent/pull/239/files#diff-61358711e980ccf505246fd3915f97cbd3a380e9b66f6fa5aad46749968c5ca3R187, so the `persist` arg was just to signal that.

We can probably get rid of the file, yes, but I don't want to go that far in this PR, just want to get the agent running standalone. :sweat_smile: